src/common/core/core.js

```diff
@@ -16,20 +19,33 @@ define([
         'common/core/metacore',
         'common/core/coretreeloader',
         'common/core/corediff'],
-    function (CoreRel, Set, Guid, NullPtr, UnWrap, Type, Constraint, CoreTree, MetaCore, TreeLoader, CoreDiff)
-{
-    "use strict";
+    function (CoreRel, Set, Guid, NullPtr, UnWrap, Type, Constraint, CoreTree, MetaCore, TreeLoader, CoreDiff) {
+    'use strict';
 
-    function core(storage,options){
-        options = options || {};
+    function core(storage, options) {
         options.usertype = options.usertype || 'nodejs'; // FIXME: why this is nodejs???
+        // = new TreeLoader(new CoreDiff(new MetaCore(new Constraint(new Guid(new Set(new NullPtr(new Type(new NullPtr(new CoreRel(new CoreTree(storage, options)))))))))));
+        var coreCon,
+            coreLayers = [];
+        coreLayers.push(CoreRel);
+        coreLayers.push(NullPtr);
+        coreLayers.push(Type);
+        coreLayers.push(NullPtr);
+        coreLayers.push(Set);
+        coreLayers.push(Guid);
+        coreLayers.push(Constraint);
+        coreLayers.push(MetaCore);
+        coreLayers.push(CoreDiff);
+        coreLayers.push(TreeLoader);
 
-        var coreCon = new TreeLoader(new CoreDiff(new MetaCore(new Constraint(new Guid(new Set(new NullPtr(new Type(new NullPtr(new CoreRel(new CoreTree(storage, options)))))))))));
+        coreCon = coreLayers.reduce(function (inner, Class) {
+            return new Class(inner, options);
+        }, new CoreTree(storage, options));
 
-        if(options.usertype === 'tasync'){
+        if (options.usertype === 'tasync') {

```

Comment:
I would put this option check before the reducing, and add Unwrap to the array if needed.